x-pack/plugins/lens/public/xy_visualization/axes_configuration.ts

```diff
@@ -33,23 +33,23 @@ export function isFormatterCompatible(
 
 export function getAxesConfiguration(
   layers: LayerConfig[],
-  tables: Record<string, KibanaDatatable>,
-  formatFactory: (mapping: SerializedFieldFormat) => IFieldFormat,
-  shouldRotate: boolean
+  shouldRotate: boolean,
+  tables?: Record<string, KibanaDatatable>,
+  formatFactory?: (mapping: SerializedFieldFormat) => IFieldFormat
 ): GroupsConfiguration {
   const series: { auto: FormattedMetric[]; left: FormattedMetric[]; right: FormattedMetric[] } = {
     auto: [],
     left: [],
     right: [],
   };
 
-  layers.forEach((layer) => {
-    const table = tables[layer.layerId];
+  layers?.forEach((layer) => {
+    const table = tables?.[layer.layerId];
     layer.accessors.forEach((accessor) => {
       const mode =
         layer.yConfig?.find((yAxisConfig) => yAxisConfig.forAccessor === accessor)?.axisMode ||
         'auto';
-      const formatter: SerializedFieldFormat = table.columns.find(
+      const formatter: SerializedFieldFormat = table?.columns.find(
```

Comment:
This logic is not correct when no table information is available because if there are two auto columns they will always be mapped to the same axis as they both end up with `{ id: 'number' }` formatter, making `isFormatterCompatible` return true.

Example visualization disabling the right axis popover even though it is active (all metrics have auto axis, with one of them with a custom formatter):
<img width="1122" alt="Screenshot 2020-09-10 at 14 15 08" src="https://user-images.githubusercontent.com/1508364/92727860-15738380-f370-11ea-9c2d-4881e10f8867.png">

One way to work around this is to change this logic:
```
      series.left.length === 0 ||
      series.left.every((leftSeries) =>
        isFormatterCompatible(leftSeries.fieldFormat, currentSeries.fieldFormat)
      )
```

to

```
      series.left.length === 0 ||
      (tables && series.left.every((leftSeries) =>
        isFormatterCompatible(leftSeries.fieldFormat, currentSeries.fieldFormat)
      ))
```

Reply:
Thank you Joe! I missed that 🙂

Reply:
About the unit tests, I would surely add but I was waiting for Caroline's feedback too